Terminal.Gui/Application/Application.Run.cs

```diff
@@ -505,6 +505,10 @@ public static void LayoutAndDraw (bool forceDraw = false)
     {
         bool neededLayout = View.Layout (TopLevels.Reverse (), Screen.Size);
 
+        if (ClearScreenNextIteration)
+        {
+            forceDraw = true;
```

Comment:
You need to set `ClearScreenNextIteration = false;` after `forceDraw = true;`, otherwise it'll always force redraw without needed.

```cs
        if (ClearScreenNextIteration)
        {
            forceDraw = true;
            ClearScreenNextIteration = false;
        }

```

Reply:
Oops. I was testing something and deleted that line! 

Fixed, and I've added some unit tests.